bebot/include/bebot/bebot.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <Eigen/Dense>
+
+namespace bebot {
+    class BeBOT {
+
+        private:
+            double _t0, _tf;
+            Eigen::MatrixXd _cpts; 
+
+        public:
+            BeBOT() = default;
+            ~BeBOT() = default;
+
+            BeBOT(const Eigen::Ref<Eigen::MatrixXd>& cpts);
+            BeBOT(const Eigen::Ref<Eigen::MatrixXd>& cpts, const double tf);
+            BeBOT(const Eigen::Ref<Eigen::MatrixXd>& cpts, const double t0, const double tf);
+
+            void set_cpts(const Eigen::Ref<Eigen::MatrixXd>& cpts);
+            Eigen::MatrixXd get_cpts();
+
+            void set_t0(const double t0);
```

Comment:
Might be a philosophy thing but unless there's a reason you expect there to be more complexity in assignment/reading, getters/setters like this are antipatterns. Even when they're not, instead of `.get_t0()`, I would prefer `.t0()`. On that note, `.initial_time` and `final_time`?